### PR TITLE
Gifting Modal Redesign

### LINKED ISSUE #12607: Update Gifting Modal Design
Need to implement the new Send Gift modals seen here: https://zpl.io/O0RPAqQ

### PR DESCRIPTION
**Pull request by CuriousMagpie**
Fixes #12607

### Changes
Update the Gifting Modal to [new design](https://zpl.io/O0RPAqQ).

- Updated `selectUserModal.vue` which users use to select the gift recipient.
- Replaced the existing `sendGemsModal.vue` with a new `sendGiftModal.vue` and redirected all links to point to the new modal, including profile page.
- Left existing `sendGemsModal.vue` in the codebase for now, but it will need to be removed.
- Ensured that the "Give One, Get One" promotion works with the new modals.
- Tested sending gems from balance; payments will need to be tested end-to-end in either staging or delta.

### CODE REVIEW

File: website/client/src/components/payments/sendGiftModal.vue [hunk 1]
```diff
@@ -0,0 +1,638 @@
...
+        <div
+          class="d-flex flex-column mx-auto align-items-center display-name">
+          <!-- user display name and username -->
+            <user-link
+              :user-id="displayName"
+              :name="displayName"
+              :backer="userBacker"
+              :contributor="userContributor"
+              :class="display-name"
+              />
```
Comment (SabreCat) [inline website/client/src/components/payments/sendGiftModal.vue]: There's a bunch of weird indentation in this file, first instance here but it's throughout--would it work for me to just proofread and commit, you think?

File: website/client/src/components/payments/sendGiftModal.vue [hunk 2]
```diff
@@ -0,0 +1,638 @@
...
+        v-show="selectedPage === 'ownGems'"
+        >
+        <div class="gem-group">
+        <label v-once>
+          {{ $t('howManyGemsSend') }}
+        </label>
+        <div class="d-flex align-items-center justify-content-center">
+          <div
+            class="gray-circle"
+            @click="gift.gems.amount <= 0 ? 0 : gift.gems.amount--"
```
Comment (SabreCat) [inline website/client/src/components/payments/sendGiftModal.vue]: In the case where `gift.gems.amount <= 0` this executes `0` as code, which I think isn't what you want? I'd replace `? 0` with `? gift.gems.amount = 0`

File: website/client/src/components/payments/sendGiftModal.vue [hunk 3]
```diff
@@ -0,0 +1,638 @@
...
+              id="gemsForm"
+              v-model="gift.gems.amount"
+              class="form-control"
+              min="0"
+              :max="maxGems"
+            >
+        </div>
+          <div
+            class="gray-circle"
+            @click="gift.gems.amount < maxGems ? gift.gems.amount++ : maxGems"
```
Comment (SabreCat) [inline website/client/src/components/payments/sendGiftModal.vue]: Similar deal here, we're executing `maxGems` (or is this some shorthand in ternaries I'm not familiar with..?)

File: website/client/src/components/payments/sendGiftModal.vue [hunk 4]
```diff
@@ -0,0 +1,638 @@
...
+  }
+
+  .nav-item {
+    display: inline-block;
+    padding: 0px 8px 6px 8px;
+  }
+
+  .nav-item:hover, .nav-item.active {
+    color: $purple-300;
+    border-bottom: 2px solid #9a62ff;
```
Comment (SabreCat) [inline website/client/src/components/payments/sendGiftModal.vue]: Interesting, this isn't one of the colors in `colors.scss` -- is it on a stylesheet in Zeplin? it looks like we use it in a handful of places... if it's not on our stylesheet, maybe check with Tressley if it ought to be; if it's on the stylesheet, let's create a color variable in `colors.scss` and employ it here.

File: website/client/src/components/payments/sendGiftModal.vue [hunk 5]
```diff
@@ -0,0 +1,638 @@
...
+
+    &:hover {
+      border-color: $purple-400;
+    }
+  }
+
+  .gray-circle:hover{
+    .icon-positive, .icon-negative {
+      & ::v-deep svg path {
+        fill: $purple-400
```
Comment (SabreCat) [inline website/client/src/components/payments/sendGiftModal.vue]: missing semicolon

File: website/client/src/components/payments/sendGiftModal.vue [hunk 6]
```diff
@@ -0,0 +1,638 @@
...
+
+// libs imports
+import { mapState } from '@/libs/store';
+
+// mixins imports
+import paymentsMixin from '../../mixins/payments';
+
+// component imports
+import avatar from '../avatar';
+import userLink from '../userLink'; // in case I need to do the tier color/icon
```
Comment (SabreCat) [inline website/client/src/components/payments/sendGiftModal.vue]: old comment?

File: website/client/src/components/payments/sendGiftModal.vue [hunk 7]
```diff
@@ -0,0 +1,638 @@
...
+        key: '',
+      },
+      icons: Object.freeze({
+        closeIcon,
+        gemIcon,
+        positiveIcon,
+        negativeIcon,
+      }),
+      userReceivingGift: {
+        profile: '', // here bc it was throwing an error
```
Comment (SabreCat) [inline website/client/src/components/payments/sendGiftModal.vue]: Comments here probably not needed either

File: website/client/src/components/settings/subscription.vue [hunk 8]
```diff
@@ -450,9 +450,9 @@
     background-color: $white;
   }
 
-  .subscribe-option {
-    border-bottom: 1px solid $gray-600;
-  }
+  // .subscribe-option {
+  //   border-bottom: 1px solid $gray-600;
+  // }
```
Comment (SabreCat) [inline website/client/src/components/settings/subscription.vue]: if not needed, delete, don't comment

File: website/common/locales/en/groups.json [hunk 9]
```diff
@@ -129,6 +129,8 @@
   "sendGiftHeading": "Send Gift to <%= name %>",
   "sendGiftGemsBalance": "From <%= number %> Gems",
   "sendGiftCost": "Total: $<%= cost %> USD",
+  "sendGiftTotal": "Total:",
+  "sendGiftAmount": "<%= cost %>",
```
Comment (SabreCat) [inline website/common/locales/en/groups.json]: This is weird, why use an entire translation string that contains only variable data? We don't need to run it through the translation function if nothing's being translated.

File: website/common/locales/en/subscriber.json [hunk 10]
```diff
@@ -196,5 +200,7 @@
     "needToUpdateCard": "Need to update your card?",
     "readyToResubscribe": "Are you ready to resubscribe?",
     "cancelYourSubscription": "Cancel your subscription?",
-    "cancelSubAlternatives": "If you're having technical problems or Habitica doesn't seem to be working out for you, please consider <a href='mailto:admin@habitica.com'>contacting us</a>. We want to help you get the most from Habitica."
+    "cancelSubAlternatives": "If you're having technical problems or Habitica doesn't seem to be working out for you, please consider <a href='mailto:admin@habitica.com'>contacting us</a>. We want to help you get the most from Habitica.",
+    "nextButton": "Next",
```
Comment (SabreCat) [inline website/common/locales/en/subscriber.json]: We already have this string as `next`

**Review by SabreCat — CHANGES_REQUESTED** [review]

File: website/client/src/components/settings/subscription.vue (hunk 8 above)
Comment (CuriousMagpie) [reply website/client/src/components/settings/subscription.vue]: deleted

File: website/common/locales/en/groups.json (hunk 9 above)
Comment (CuriousMagpie) [reply website/common/locales/en/groups.json]: changed the component, removing line 133.

File: website/common/locales/en/subscriber.json (hunk 10 above)
Comment (CuriousMagpie) [reply website/common/locales/en/subscriber.json]: updated!

File: website/client/src/components/payments/sendGiftModal.vue (hunk 3 above)
Comment (CuriousMagpie) [reply website/client/src/components/payments/sendGiftModal.vue]: We need to execute `maxGems` in order to get the user's total number of gems to compare with the `gift.gems.amount`--unless there's something I'm missing here?

File: website/client/src/components/payments/sendGiftModal.vue (hunk 5 above)
Comment (CuriousMagpie) [reply website/client/src/components/payments/sendGiftModal.vue]: fixed

File: website/client/src/components/payments/sendGiftModal.vue (hunk 6 above)
Comment (CuriousMagpie) [reply website/client/src/components/payments/sendGiftModal.vue]: yep!

File: website/client/src/components/payments/sendGiftModal.vue (hunk 7 above)
Comment (CuriousMagpie) [reply website/client/src/components/payments/sendGiftModal.vue]: I put these in so someone looking at the code in the future would know why we were passing empty strings to those variables but I can definitely take them out.

File: website/client/src/components/payments/sendGiftModal.vue (hunk 4 above)
Comment (CuriousMagpie) [reply website/client/src/components/payments/sendGiftModal.vue]: It's supposed to be $purple-400, was an artifact,

**Review by SabreCat — APPROVED** [review]

**Review by SabreCat — APPROVED** [review]

**Pull request merged**